### ISSUE TITLE
0.20 package/paths resolution issue

### ISSUE THREAD

**Issue by ghempton** [systemjs/systemjs#1552]
When using previous version of SystemJS, the configuration looks like this:

```
SystemJS.config({
  paths: {
    "app/": "modules/app/"
  },
  packages: {
    "app": {
      "format": "register",
      "defaultExtension": "js"
    }
});
```

After upgrading to 0.20, the `app` package configuration seems to no longer apply to modules within the package and it does not apply the `defaultExtension`.

```
System.import('app/start');

// console error saying it cannot find 'https://localhost:4200/modules/app/start'
```

If instead I change the configuration to:

```
SystemJS.config({
  paths: {
    "app/": "modules/app/"
  },
  packages: {
    "modules/app": {
      "format": "register",
      "defaultExtension": "js"
    }
});
```

It starts to work, but then importing the base module fails:

```
System.import('app');
```

It looks like for non-root modules, it is applying path configuration before looking up the package. Is this expected behavior?

**Comment by patrickmichalina** [comment]
Same issue here. Had to revert to previous version.

**Comment by guybedford** [comment]
Thanks, I've added a fix in https://github.com/systemjs/systemjs/commit/83111deafbbe7d773aa8cbf4bd27ea6d6b652e50.

**Comment by guybedford** [comment]
Released in 0.20.2.
